src/common/Types.ts

```diff
@@ -835,7 +881,7 @@ export interface LinkProps extends CommonStyledProps<LinkStyleRuleSet> {
 export interface TextInputPropsShared extends CommonProps, CommonAccessibilityProps {
     autoCapitalize?: 'none' | 'sentences' | 'words' | 'characters';
     autoCorrect?: boolean;
-    autoFocus?: boolean;
+    autoFocus?: AutoFocus | AutoFocus[]; // Should autofocus depending on the set of the specified conditions
```

Comment:
Changing the type of this prop will result in a breaking change to the existing interface. There needs to be a _really_ compelling reason to do so, and I don't think there is in this case. We should continue to support the old prop but also include support for a new prop (with a different name to avoid conflict).